mex/common/settings.py

```diff
@@ -10,8 +10,7 @@
 from mex.common.context import ContextStore
 from mex.common.types import AssetsPath, IdentityProvider, Sink, WorkPath
 
-SettingsType = TypeVar("SettingsType", bound="BaseSettings")
-SettingsContext = ContextStore[Optional["BaseSettings"]](None)
+SettingsContext = ContextStore[dict[type["BaseSettings"], "BaseSettings"]]({})
```

Comment:
short answer: `ContextStore` already was a generic class that was parameterized by a typevar (`ContextResource`) and you can't have a typevar inside a typevar. but it still works because `BaseSettings` are just the upper bound here. long answer: [covariance and contravariance](https://peps.python.org/pep-0484/#covariance-and-contravariance).

third answer: your comment led me to refactoring the `ContextStore` thing entirely to (hopefully) have a more intuitive naming and API: https://github.com/robert-koch-institut/mex-common/pull/139/commits/c060d84b76bea862e5030dc32fee6a3940eaad2d

Reply:
neat!